addons/account/models/account_move.py

```diff
@@ -955,6 +955,10 @@ def _compute_currency_id(self):
             )
             invoice.currency_id = currency
 
+    def _get_invoice_currency_rate_date(self):
+        self.ensure_one()
+        return self.invoice_date
```

Comment:
```suggestion
        return self.invoice_date or fields.Date.context_today(self)
```
Maybe we could handle all the logic inside this method?